### PR TITLE
Refactor extractors

### PR DESCRIPTION
**Pull request by enolfc**
Add StAR support with XML records generated
Fix some minor issues found while testing

### CODE REVIEW

**Comment by enolfc** [comment]
Any feedback on this one? It would be good to have some records sent asap to test things work as expected

**Comment by aidaph** [comment]
I sent my comments 8 days ago, so weird that can't you see them? I copy my review below:

Testing the code, there is a mess between the datetime types. Some of them are with TZ INFO and others with TZ INFO = None. I've got an Unhandled error: TypeError: can't compare offset-naive and offset-aware datetimes

for example:
if extract_to > now:
2022-07-21 13:57:34.542 8871 ERROR caso TypeError: can't compare offset-naive and offset-aware datetimes

As I see debugging the values, the datetime types mismatch:
[....] #### Extract_to: 2022-07-21 11:57:34.541254+00:00 and now: 2022-07-21 11:57:34.541254 get_records /gpfs/users/aidaph/test_caso/lib/python3.6/site-packages/caso/extract/manager.py:121

**Comment by enolfc** [comment]
ups I didn't see any comments at all... 

anyway. I also had lots of issues with the TZ, but it seems we are not looking at the same code, as this line 121 is not the comparison that may fault. Can you check you have the latest code?

**Review by aidaph — APPROVED** [review]

File: caso/record.py [hunk 1]
```diff
@@ -277,25 +279,27 @@ def validate_attached_duration(cls, value):
...
-                'user_id': 'LocalUser',
+                "user_id": "LocalUser",
                 "user_dn": "GlobalUserName",
-                'group_id': 'LocalGroup',
+                "group_id": "LocalGroup",
                 "fqan": "FQAN",
                 "site_name": "SiteName",
                 "capacity": "Capacity",
                 "active_duration": "ActiveDuration",
+                "start_time": "ActiveDuration",
```
[the remark this answers is not in the document]
Comment (enolfc) [reply caso/record.py:296]: just got the comment now

[the remark this answers is not in the document]
Comment (enolfc) [reply caso/record.py:296]: > StartTime instead of ActiveDuration?

yup, fixing this

**Review by aidaph — APPROVED** [review]

**Pull request merged**